module/VuFind/src/VuFind/Search/Base/Params.php

```diff
@@ -747,14 +747,8 @@ public function setSort($sort, $force = false)
         // Validate and assign the sort value:
         $valid = array_keys($this->getOptions()->getSortOptions());
 
-        $matchedHiddenPatterns = array_filter(
-            $this->getOptions()->getHiddenSortOptions(),
-            function ($pattern) use ($sort) {
-                return preg_match('/' . $pattern . '/', $sort);
-            }
-        );
-
-        if (!empty($sort) && (in_array($sort, $valid) || count($matchedHiddenPatterns) > 0)) {
+        $matchedHiddenPatterns = $this->getMatchingHiddenSortingPatterns($sort);
+        if (!empty($sort) && (in_array($sort, $valid) || $matchedHiddenPatterns)) {
```

Comment:
I think this would be a little more efficient -- avoid processing patterns unless we really need to:

```suggestion
        if (
            !empty($sort)
            && (in_array($sort, $valid) || $this->getMatchingHiddenSortingPatterns($sort))
        ) {
```